arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -107,7 +108,13 @@ RequestLane RestDocumentHandler::lane() const {
 RestStatus RestDocumentHandler::execute() {
   // extract the sub-request type
   auto const type = _request->requestType();
-
+#ifdef ARANGODB_ENABLE_FAILURE_TESTS
+  std::vector<std::string> const& suffixes = _request->decodedSuffixes();
+  TRI_IF_FAILURE("failOnCRUDAPI" + suffixes[0]) {
```

Comment:
Is accessing `suffixes[0]` unconditionally safe here? What if someone just calls `/_api/document` without any suffixes?
I think it is safer to put the `TRI_IF_FAILURE...` into an `if (!suffixes.empty()) { ... }` condition.

Reply:
Ah good point. The if-def will be there in all of our Test Builds. So it may have undesired side effects. Will fix!